File: 3.py
# 插入排序
def insert_sort(arry):
    n = len(arry)
    for i in range(1,n):
        if arry[i] < arry[i-1]:
            temp = arry[i]
            index = 1
            for j in range(i-1,-1,-1):
                if arry[j] > temp:
                    arry[j+1] = arry[j]
                    index = j
                else:
                    break
                arry[index] = temp
    return arry

File: 3.cs
//插入排序
public static void InsertSort(int[] list)
{
  for (int i = 1; i < list.length; i++)
  {
    int temp=list[i];
    int j = i-1;
    while (j>=0 && temp < list[j]);
    {
      list[j+1] = list[j];
      j--;
    }
    list[j+1] = temp;
  }
}
